src/feature/control/control_fmt.c

```diff
@@ -165,6 +171,74 @@ circuit_describe_status_for_controller(origin_circuit_t *circ)
   return rv;
 }
 
+/**
+ * Extends the description of a <b>circ</b>'s status.
+ */
+void
+circuit_extend_status_for_controller(const origin_circuit_t *circ,
+                                     smartlist_t *descparts)
+{
+  // Supplementary data
+  smartlist_add_asprintf(descparts, "UNUSABLE_FOR_NEW_CONNS=%u",
+                         circ->unusable_for_new_conns);
+  smartlist_add_asprintf(descparts, "IDLE_TIMEOUT=%d",
+                         circ->circuit_idle_timeout);
+  smartlist_add_asprintf(descparts, "EARLY_CELLS_SENT=%u",
+                         circ->relay_early_cells_sent);
+  smartlist_add_asprintf(descparts, "REMAINING_RELAY_EARLY_CELLS=%u",
+                         circ->remaining_relay_early_cells);
+  smartlist_add_asprintf(descparts, "PADDING_NEGOTIATION_FAILED=%u",
+                         circ->padding_negotiation_failed);
+  smartlist_add_asprintf(descparts, "PATH_STATE=%s",
+                         pathbias_state_to_controller_string(circ->path_state)
+                         );
+  smartlist_add_asprintf(descparts, "HAS_RELAXED_TIMEOUT=%u",
+                         circ->relaxed_timeout);
+  smartlist_add_asprintf(descparts, "IS_ANCIENT=%u", circ->is_ancient);
+  smartlist_add_asprintf(descparts, "HAS_OPENED=%u", circ->has_opened);
```

Comment:
Should %u really be used as a format string here for these boolean types? It might be we do that elsewhere and it's fine, but it just caught my eyes as being odd. %u can have a lot of different values other than 0 or 1, but the spec also doesn't allow us to use the "true" and "false" literal.

Reply:
I have to admit that I do not really know how to print those values otherwise. They are defined as unsigned int, so I printed them as such, even if they are mostly defined on 1 bit.

I tried to look in the project, but I could not see another boolean output definition. Do you have something in mind to use instead of `%u`?

Reply:
Hello,

I'm coming back to you on this part. I personally don't feel like it's an issue to keep `%u` since the fields are effectively `unsigned int` limited to 1 bit (Cf. https://github.com/torproject/tor/blob/master/src/core/or/origin_circuit_st.h#L151-L156).

However, if you prefer using `true` and `false` literals, I can change the specs on this part since it hasn't been merged already.

What would you prefer? :)